feat: profile social links

flagrede opened the pull request:
## Description

Closes: regen-network/regen-registry#1616
registry-server PR: https://github.com/regen-network/registry-server/pull/311

Add social links to the user profile.

Note that:
- some parts of this issue have been done in the edit profile PR in this [commit](https://github.com/regen-network/regen-web/pull/1826/commits/8980e758e89c10922a2aaf1b26be4cd95e84823b).
- preview will be usable once registry-server PR is merged

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. https://deploy-preview-1839--regen-registry.netlify.app/profile/edit

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

haveanicedavid reviewed (approved):
lgtm

blushi commented:
could you rebase this from dev so it has the latest changes from #1819?

blushi reviewed (commented):
"Links" label is using Muli as font somehow instead of Lato
![image](https://user-images.githubusercontent.com/3149358/228766575-d58a0bc0-745d-4e21-953c-de8a91a9c19b.png)

flagrede commented on web-registry/src/components/organisms/EditProfileForm/EditProfileForm.constants.tsx:
```diff
@@ -41,7 +41,7 @@ export const radioCardItems: RadioCardItem[] = [
-export const PROFILE_BG_ASPECT_RATIO = 4;
+export const PROFILE_BG_ASPECT_RATIO = 1440 / 326;
 export const LINKS_LABEL = 'Links';
 export const WEBSITE_PLACEHOLDER = 'yourwebsite.com';
 export const TWITTER_PLACEHOLDER = 'yourtwitterhandle';
```
Good catch, probably forgot to use them.

flagrede commented:
> "Links" label is using Muli as font somehow instead of Lato ![image](https://user-images.githubusercontent.com/3149358/228766575-d58a0bc0-745d-4e21-953c-de8a91a9c19b.png)

@blushi that's weird, will fix it, thanks.

flagrede commented:
@erikalogie you can now test the complete profile feature!

I might have found a small design issue. If the user updates an image with a white background, the button is no longer visible. We might need to add a small dark background to this button, wdyt?

erikalogie commented:
@flagrede Is an error expected if I click the "edit" button?

Yes, actually I just looked at the comps and realized that I had put a 50% black overlay on the background image. So try adding that, I may need to provide you with a different default image if that looks now looks too dark with the overlay.

One more thing that I noticed is that we want to use the grey version of the org icon on profile page: https://monosnap.com/file/5p4ejZXTZZVCO0IBE9uiJ6GJKDBPGg
![Organization-grey](https://user-images.githubusercontent.com/8460237/228874577-1d390064-f050-4699-9983-b97f79e71177.svg)

Also can you fix this spacing? 
![image](https://user-images.githubusercontent.com/8460237/228876254-848d560d-84b3-499a-b0cd-63ebda081f60.png)

flagrede commented:
@erikalogie Sorry that might be because I was doing some test at the same time, could you give it another try? Everything should be working, I just double-checked.

flagrede commented:
@erikalogie I handled all your feedback, here are some notes on them:
- For the black overlay at 50% the image felt a bit too dark with it and also it was a shame to apply it to the whole image. I tried instead using a black gradient that follows the position of the text based on the breakpoint and set it to 20% opacity. Let me know what you think, we can increase the opacity a bit more if you feel the contrast is insufficient. If you want to check the linear background separately you can see it here: https://deploy-preview-1839--regen-storybook.netlify.app/?path=/story/organisms-profileheader--white-bg
- I added the validation check for the URLs, the default message is "Invalid URL", let me know if you want to change that. Also, I changed the Twitter placeholder from `yourtwitterhandle` to `twitter.com/yourtwitterhandle` because I find it confusing otherwise, I can revert that if you think it's not necessary.

erikalogie commented:
> @erikalogie I handled all your feedback, here are some notes on them:
> 
> * For the black overlay at 50% the image felt a bit too dark with it and also it was a shame to apply it to the whole image. I tried instead using a black gradient that follows the position of the text based on the breakpoint and set it to 20% opacity. Let me know what you think, we can increase the opacity a bit more if you feel the contrast is insufficient. If you want to check the linear background separately you can see it here: https://deploy-preview-1839--regen-storybook.netlify.app/?path=/story/organisms-profileheader--white-bg

This looks great!

> * I added the validation check for the URLs, the default message is "Invalid URL", let me know if you want to change that. Also, I changed the Twitter placeholder from `yourtwitterhandle` to `twitter.com/yourtwitterhandle` because I find it confusing otherwise, I can revert that if you think it's not necessary.

Good call. I actually think we should change the dummy text to http://yourwebsite.com and http://twitter.com/yourtwitterhandle since it throws an error without the https:// unless there is an easy way to detect and append to the url without the user input.

blushi reviewed (approved):
pre-approving, pending letter spacing issue for the "Links" label

blushi commented:
Also shouldn't both social links be optional? cc/ @erikalogie @flagrede

flagrede commented:
@blushi I think you're right, this should be optional.

flagrede commented:
@blushi I just checked, they were already optional, just missing the text on the label, will add it.
